portal/src/main/webapp/js/api/cbioportal-datamanager.js

```diff
@@ -245,9 +328,118 @@ window.initDatamanager = function (genetic_profile_ids, oql_query, cancer_study_
 	/* in-place, idempotent
 	 * In: - webservice_data, a list of data obtained from the webservice API
 	 * Out: promise, which resolves with the data which has been in-place modified,
-	 *	    the mutation data given the boolean attribute 'oncokb_oncogenic'
+	 *	    the mutation data given the string attribute 'oncokb_oncogenic', one of ['Unknown', 'Likely Neutral', 'Likely Oncogenic', 'Oncogenic']
 	 */
+	var def = new $.Deferred();
+	var attribute_name = 'oncokb_oncogenic';
+	getOncoKBAnnotations(webservice_data).then(function(oncogenic) {
+	    for (var i=0; i<webservice_data.length; i++) {
+		var datum = webservice_data[i];
+		if (datum.genetic_alteration_type !== "MUTATION_EXTENDED") {
+		    continue;
+		}
+		var gene = datum.hugo_gene_symbol;
+		gene && (gene = gene.toUpperCase());
+		var alteration = datum.amino_acid_change;
+		alteration && (alteration = alteration.toUpperCase());
+		if (gene && alteration && oncogenic[gene] && oncogenic[gene][alteration]) {
+		    datum[attribute_name] = oncogenic[gene][alteration];
+		}
+	    }
+	    def.resolve(webservice_data);
+	});
+	return def.promise();
+    };
+    var annotateCOSMICCount = function(webservice_data) {
+	/* in-place, idempotent
+	 * In: - webservice_data, a list of data obtained from the webservice API
+	 * Out: promise, which resolves with the data which has been in-place modified,
+	 *	    the mutation data given the string attribute 'cosmic_count'
+	 */
+	var def = new $.Deferred();
+	var attribute_name = 'cosmic_count';
 	
+	getCOSMICCounts(webservice_data).then(function(cosmic_counts) {
+	    for (var i=0; i<webservice_data.length; i++) {
+		var datum = webservice_data[i];
+		if (datum.genetic_alteration_type === "MUTATION_EXTENDED" && typeof cosmic_counts[datum.keyword] !== "undefined") {
+		    var count_records = cosmic_counts[datum.keyword];
+		    // Filter by position if 'truncating'
+		    if (datum.keyword.indexOf("truncating") > -1) {
+			var protein_start_position = parseInt(datum.protein_start_position, 10);
+			count_records = count_records.filter(function(count_record) {
+			    return count_record.protein_change && parseInt(count_record.protein_change, 10) === protein_start_position;
+			});
+		    }
+		    datum[attribute_name] = count_records.map(function(count_record) { return parseInt(count_record.count, 10); })
+							.reduce(function(x,y) { return x+y; }, 0);
+		}
+	    }
+	    def.resolve(webservice_data);
+	});
+	return def.promise();
+    };
+    
+    var annotateHotSpots = function(webservice_data) {
```

Comment:
Why don't we make it as an abstract utility function, so that it can be invoked from other views if we require the same functionality.